platform/darwin/src/MGLLoggingConfiguration.mm

```diff
@@ -1,14 +1,60 @@
+#include <mbgl/util/logging.hpp>
+
 #import "MGLLoggingConfiguration_Private.h"
 
 #ifndef MGL_LOGGING_DISABLED
 #if __has_builtin(__builtin_os_log_format)
 #import <os/log.h>
 #endif
 
-@implementation MGLLoggingConfiguration
+namespace mbgl {
+    
+class MGLCoreLoggingObserver : public Log :: Observer {
+public:
+    //Filter logs from core
+    bool onRecord(EventSeverity severity, Event, int64_t, const std::string&) override{
+        if(_platformLoggingLevel == MGLLoggingLevelNone){
```

Comment:
We also need to pass the message to our logging handler in this function. 😄

Reply:
A question: why we need to pass the message to our logging handler? Since the message has already printed by `logging_apple.mm`(previously `logging_nslog.mm`), I intercept these messages by `MGLCoreLoggingObserver` - when `onRecord` function return true, the message won't be printed.

Reply:
> Let's change this to a switch. Or... this could be an array test with bit masks, something along the lines of:
> 
> ```
> return (coreEventSeverityMask[self.platformLoggingLevel] & (1 << severity))
> ```

Actually, firstly I already try to use a bit mask like:
```
    /*
        None                       0
        EventSeverity::Debug       1 << 0
        EventSeverity::Info        1 << 1
        EventSeverity :: Warning   1 << 2
        EventSeverity::Error       1 << 3
     */
    constexpr int static coreEventSeverityMask[7] = {
                                                        0,
                                                        1 << 3,
                                                        1 << 3,
                                                        1 << 2 | 1 << 3,
                                                        1 << 1 | 1 << 2 | 1 << 3,
                                                        1 << 0,
                                                        1 << 0,
                                                    };
```

However, the `severity` is a `enum class`, which means I cannot make an implicit conversion and make a bitwise operation like `1 << severity`. That's why I wrote down so many conditions. @julianrex

Reply:
> why we need to pass the message to our logging handler? Since the message has already printed by logging_apple.mm

When `onRecord` returns `true` the message is *not* printed by `platformRecord`. The idea is to pass the logging message from core, all the way through to the application.

Reply:
> which means I cannot make an implicit conversion and make a bitwise operation like 1 << severity

Perhaps we can use an inline constexpr function here?

```c++
constexpr unsigned int mask(const EventSeverity severity) {
    return (1 << (unsigned int)severity);
}
```